leader/leader.go

```diff
@@ -212,16 +212,19 @@ func (l *Client) AddWatch(key string, retry time.Duration, valuesC chan string)
 	}()
 }
 
-// AddVoter adds a voter that tries to elect given value
-// by attempting to set the key to the value for a given term duration
-// it also attempts to hold the lease indefinitely
-func (l *Client) AddVoter(key, value string, term time.Duration) error {
+// AddVoter starts a goroutine that attempts to set the specified key to
+// to the given value with the time-to-live value specified with term.
+// The time-to-live value cannot be less than a second.
+// After successfully setting the key, it attempts to renew the lease for the specified
+// term indefinitely
+func (l *Client) AddVoter(key, value string, term time.Duration) (io.Closer, error) {
 	if value == "" {
-		return trace.Errorf("voter value for key can not be empty")
+		return nil, trace.Errorf("voter value for key cannot be empty")
 	}
 	if term < time.Second {
-		return trace.Errorf("term can not be < 1second")
+		return nil, trace.Errorf("term cannot be < 1s")
 	}
+	voter := &voter{doneC: make(chan struct{})}
```

Comment:
let's use `context.Done()` that is reentrable instead of closing channel that will panic on multi-close